### DISCUSSION TITLE
Using iVABS without virtual environment

### DISCUSSION THREAD

**Discussion by Si1verBe11** [wenbinyugroup/.github#17] [VABS Related Discussions] [answered]
I'd like to use iVABS, but I can't not use virtual environment because of company's policy.
Instead of that, I just downloaded potable iVABS v0.11 and tried to run, but failed.

I already installed all required Python packages.
![image](https://github.com/user-attachments/assets/e377d57c-aa46-4865-b3a5-40057883a833)

And set up PATH and PYTHONPATH
![image](https://github.com/user-attachments/assets/75a71edd-b5d7-470b-8b14-2252d5a57c3d)

But if I type `ivabs`, I got this:
![image](https://github.com/user-attachments/assets/d6ba087d-d27a-4ce1-986e-40f77b66ce8d)

Is this a problem because I'm not using a virtual environment? Or is it something else?
If virtual environment is the problem, can I use iVABS without it?

**Comment by unitasium** [comment]
This is caused by some changes in the numpy 2.0. This will be resolved in a future version. Right now the way to fix this is to install an earlier version: `pip install numpy<2.0`.

**Reply by Si1verBe11** [reply]
I installed the last version of Numpy v1.X. And it works!
![image](https://github.com/user-attachments/assets/230db85e-dd26-4a76-a447-70e2b8ac2a45)

But I got another problem like this:
![image](https://github.com/user-attachments/assets/4f6f241e-e1c7-42f3-85b5-893237e91e18)

So, I downloaded old version of PreVABS(https://cdmhub.org/resources/1597/supportingdocs).
And add `PREVABS_DIR` to the environment variable `PATH`.
Also, I temporarily renamed the prevabs.exe in the `%iVABS_dir%/bin` so that it doesn't work (prevabs_ivabs.exe)
Then it worked fine:
![image](https://github.com/user-attachments/assets/0e7eaaac-94cf-4403-af8d-71d3aecbf806)

It sounds like there is an issue with the prevabs.exe included in the iVABS folder, which would be good to check.

**Reply by Si1verBe11** [reply]
While working through the example, I additionally noticed the following errors occurring
This is an example of optimizing a laminate for a box beam, but it doesn't work.
![image](https://github.com/user-attachments/assets/e504b206-e735-44ce-9939-f1cd6d60ccfa)

**Reply by unitasium** [reply]
Can you run `ivabs --loglevelfile debug iter main.yml` and show the message in `evals/eval.1/eval.log`?

**Reply by Si1verBe11** [reply]
Here I attached the `evals/eval.1/eval.log` : [eval.log](https://github.com/user-attachments/files/17138160/eval.log)

**Reply by unitasium** [reply]
Can you try this command in `evals/eval.1`: `prevabs -i cs\main_cs_set1.xml -vabs -ver 4.1 -h`?

**Reply by Si1verBe11** [reply]
This is the results:
![image](https://github.com/user-attachments/assets/b3b72a8a-a281-4c6b-b878-49db3487382b)

Two things,
1. The prevabs in the downloaded iVABS folder didn't work, so I downloaded PreVABS 1.5.1 separately and am using it. This was already explained above (https://github.com/orgs/wenbinyugroup/discussions/17#discussioncomment-10754255).
2. The current message says that there is no lamina named la_mat_1, which I have confirmed does not exist in the MaterialDB.xml included in PreVABS 1.5.1, nor in the MaterialDB.xml included in iVABS 0.11.

**Reply by unitasium** [reply, marked as answer]
The examples are tested against the latest versions delivered in the ivabs package, so there could be some compatibility issues with older versions of prevabs. The issue of the latest prevabs (1.5.2) not running on your machine could be related with the mismatch of vabs dll. The code is compiled along with the library of vabs 4.0. If you are using vabs 4.1, then it may not work. I am working on updating prevabs to fix this.

The material and lamina data are stored in the main cross-section input file (`opt_cs_box_lam/files/box.xml.tmp`). The older version of prevabs looks for material data from an external database only. This is why it cannot find the lamina. The workaround is to copy the material and lamina data from the main input (`box.xml.tmp`) to the database (`MaterialDB.xml`).
